Geometry/HcalAlgo/plugins/dd4hep/DDHCalEndcapModuleAlgo.cc

```diff
@@ -82,7 +82,8 @@ struct HCalEndcapModuleAlgo {
           zpos(z) {}
   };
 
-  HCalEndcapModuleAlgo() {}
+  HCalEndcapModuleAlgo() { throw cms::Exception("HCalGeom") << "Wrong initialization to HcalEndcapModuleAlgo"; }
```

Comment:
it is better to delete the default constructor so unwanted behavior can be noticed at compile time rather than run time:
```
HCalEndcapModuleAlgo() = delete;
```